app/src/main/java/com/ddoczi/tasky/agenda/presentation/detail/agenda/AgendaDetailScreen.kt

```diff
@@ -36,18 +38,33 @@ import java.time.LocalDate
 fun AgendaDetailScreen(
     state: AgendaDetailState,
     onEvent: (AgendaDetailEvent) -> Unit,
-    agendaType: AgendaType = AgendaType.TASK
+    agendaType: AgendaType,
+    agendaOption: AgendaOption?
 ) {
+    val editTitle = stringResource(id = R.string.edit_title)
+    val editDesc = stringResource(id = R.string.edit_description)
+
+    LaunchedEffect(agendaType) {
+        onEvent(AgendaDetailEvent.OnInitScreen(agendaType, agendaOption))
+    }
```

Comment:
Is there a specific reason you have to call this init function from the screen? That will always fire again when there's a config change. Depending on what you're trying to do, you might rather want to do this in the `init` block of your ViewModel which is only called once per screen.

To get navigation arguments in a ViewModel, you can pass a `SavedStateHandle` instance to its constructor (Hilt will auto-inject it) and then retrieve navigation parameters from it with `savedStateHandle.get<String>("your_string_param")`